cli/k8s_client/yaml_factory.go

```diff
@@ -433,6 +433,16 @@ func GetCSIDeploymentYAML(args *DeploymentYAMLArguments) string {
 		autosupportDebugLine = "#" + autosupportDebugLine
 	}
 
+	if args.CloudProvider == CloudProviderAzure {
```

Comment:
`strings.EqualsFold`

and also could you provide an example how to set this `cloudProvider`

Reply:
I added a value named cloudProvider in `helm/trident-operator/values.yaml`. Deploy with helm option `--set cloudProvider=`.